modules/react-components/src/components/dropdown/app-switcher/app-switcher.tsx

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TestableComponentInterface } from "@wso2is/core/models";
+import classNames from "classnames";
+import React, { FunctionComponent, ReactElement, ReactNode, SyntheticEvent } from "react";
+import { Dropdown, DropdownItemProps, DropdownProps, Item } from "semantic-ui-react";
+import NineDotsIcon from "../../../assets/images/icons/nine-dots-icon.svg";
+import { GenericIcon, GenericIconProps } from "../../icon";
+import { Tooltip, TooltipPropsInterface } from "../../typography";
+
+/**
+ * App Switcher component props interface.
+ */
+export interface AppSwitcherPropsInterface extends TestableComponentInterface, DropdownProps {
+
+    /**
+     * Set of apps in the switcher
+     */
+    apps: AppSwitcherItemPropsInterface[];
+    /**
+     * Is app switcher enabled.
+     */
+    enabled?: boolean;
+    /**
+     * Tooltip.
+     */
+    tooltip?: ReactNode;
+    /**
+     * Tooltip extra props.
+     */
+    tooltipOptions?: TooltipPropsInterface;
+    /**
+     * Override the default Trigger icon.
+     */
+    triggerIcon?: ReactNode;
+}
+
+export interface AppSwitcherItemPropsInterface extends TestableComponentInterface, DropdownItemProps {
+
+    /**
+     * Short description about the app.
+     */
+    description: string;
```

Comment:
Can we make the description optional, WDYT?

Reply:
We can't maintain consistent UI if we make this optional. Some apps will have a description and some will not, which might result in empty spaces in the dropdown.